### PR TITLE
Vaishnavi_Assignment_19_80MIN_Done

### PR DESCRIPTION
**Pull request by VaishnaviVJun20**
image
image
image
image
image

### CODE REVIEW

File: src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package vaishnaviVJun20_arrayExamples;
+
+public class FirstDifferIndex {
+	
+	void findFirstDifferIndex(int[] array1, int[] array2) {
+	   for(int index=0;index<array1.length;index++) {
+		   if(array1[index]!= array2[index])
+			   System.out.println("Values are not matching at index->" +index);
```
Comment (KrishnaKTechnocredits) [inline src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java]: Where is the return statement ?

File: src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	   }	
+	}
+
+	public static void main(String[] args) {
+		FirstDifferIndex firstDifferIndex = new FirstDifferIndex();
+		int[] array1 = {10,2,9,14,3};
+		int[] array2 = {10,2,18,14,3};
```
Comment (KrishnaKTechnocredits) [inline src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java]: If here arrays value changed then your code is not working without return statement.

File: src/vaishnaviVJun20_arrayExamples/Prime.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			 flag= false;
+			 System.out.println(input+" is not a prime number");
+			 break;
+		   }
+		}
+		if(flag == true)
+			System.out.println(input +" is a Prime number");
+	}
+	
+	void findPrimeNumbers() {
```
Comment (KrishnaKTechnocredits) [inline src/vaishnaviVJun20_arrayExamples/Prime.java]: This method should be parameterized. If tomorrow I want to use this method for range 100 to 1000, this won't work.

File: src/vaishnaviVJun20_arrayExamples/Prime.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		    }
+		    if(flag == true)
+		    	sum +=index;
+		    			    
+		 } 
+		 System.out.println("count of prime numbers between 2 to 100 is: " +sum);
+		
+	}
+	
+	void findAvgOfAllPrimeNumbers() {
```
Comment (KrishnaKTechnocredits) [inline src/vaishnaviVJun20_arrayExamples/Prime.java]: No need to write extra methods, Just create two instance variable sum and count and perform that operation in isPrime method and use those variable in method like:
void findPrimeNumberInGivenRange(int startNumber, int endNumber) {
		for(int index=startNumber; index<=endNumber; index++) {
			verifyNumberIsPrimeOrNot2(index);
		}
		System.out.println("Total count is: "+count);
		System.out.println("Sum of all Prime number is: "+sum);
		System.out.println("Average of all prime number is: "+sum/count);

**Review by KrishnaKTechnocredits — CHANGES_REQUESTED** [review]
.

File: src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java (hunk 1 above)
Comment (VaishnaviVJun20) [reply src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java]: I have added the return statement. Got the logic. Thanks for the suggestion.

File: src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java (hunk 2 above)
Comment (VaishnaviVJun20) [reply src/vaishnaviVJun20_arrayExamples/FirstDifferIndex.java]: Done

File: src/vaishnaviVJun20_arrayExamples/Prime.java (hunk 3 above)
Comment (VaishnaviVJun20) [reply src/vaishnaviVJun20_arrayExamples/Prime.java]: Done suggested changes.

File: src/vaishnaviVJun20_arrayExamples/Prime.java (hunk 4 above)
Comment (VaishnaviVJun20) [reply src/vaishnaviVJun20_arrayExamples/Prime.java]: done suggested changes. Thanks for the logic suggestion.

**Review by KrishnaKTechnocredits — APPROVED** [review]
Good job dear.

**Pull request merged**